src/app/filecoin-plus/allocators/utils/getAllocatorsWithDatacap.ts

```diff
@@ -0,0 +1,21 @@
+import { getAllocators } from '../utils/getAllocators'
+import { getDatacapStats } from '../utils/getDatacapStats'
+
+export async function getAllocatorsWithDatacap() {
+  const allocators = await getAllocators()
+  const datacapStats = await getDatacapStats()
+
+  const allocatorsWithDatacap = allocators.map((allocator) => {
+    const stats = datacapStats.find(
+      (stats) => stats.address === allocator.address,
+    )
```

Comment:
The change replaces `find()` with a Map lookup. While `find()` has to search through the entire array each time, Map provides instant lookups by address. This makes the code more efficient, especially with larger datasets, since we only build the Map once instead of searching repeatedly.

```suggestion
  const statsMap = new Map(datacapStats.map((stats) => [stats.address, stats]))
```

Reply:
Done.